src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java

```diff
@@ -364,6 +385,81 @@ public boolean attackEntityFrom(@NotNull final DamageSource damageSource, final
         return result;
     }
 
+    /**
+     * Run away from an attacker
+     *
+     * @param attacker the attacking Entity
+     */
+    private void performMoveAway(@Nullable final Entity attacker)
+    {
+        this.getCitizenStatusHandler().setLatestStatus(new TextComponentTranslation("com.minecolonies.coremod.status.avoiding"));
+
+        // Environmental damage
+        if (!(attacker instanceof EntityLivingBase))
+        {
+            moveAwayPath = this.getNavigator().moveAwayFromEntityLiving(this, 5, INITIAL_RUN_SPEED_AVOID);
+            return;
+        }
+
+        // Makes the avoidance AI take over.
+        currentlyFleeing = true;
+
+        if ((getCitizenJobHandler().getColonyJob() instanceof AbstractJobGuard))
```

Comment:
Should the guards also move away?

Reply:
guards wont move away, as they do not have the moveawayAI added at all.

Reply:
moveAwayPath = this.getNavigator().moveAwayFromEntityLiving(this, 5, INITIAL_RUN_SPEED_AVOID)

but it does this for the guards, isn't that unnecessary overhead?

Reply:
no this is intended, it is running away a little bit from the current pos when getting hurt by environmental, nonentity caused damage